src/ripple/app/ledger/LedgerConsensus.h

```diff
@@ -54,6 +54,37 @@ class LedgerConsensus
 
     virtual bool peerPosition (LedgerProposal::ref) = 0;
 
+    /** Apply open ledger transactions to a view.
+
+        Preconditions:
+
+            Caller must have ownership of the master mutex.
```

Comment:
@rec It would be great if you could fetch this branch and study the call chains to verify that the invariants are enforced